core/src/main/scala/akka/kafka/javadsl/Producer.scala

```diff
@@ -31,6 +33,20 @@ object Producer {
       .asJava
 
   /**
+   * The `plainSink` can be used for publishing records to Kafka topics.
+   * The `record` contains a topic name to which the record is being sent, an optional
+   * partition number, and an optional key and value.
+   */
+  def plainSink[K, V](
+    producerProvider: () => KafkaProducer[K, V],
```

Comment:
I'm sorry, I don't understand the problem. Here is a [patch](https://gist.github.com/patriknw/5f9eebb86410d281741b3f4734b96d34) that shows what I would like. It simply removes the `() =>` from the API.

I think that is a better API since it conveys the purpose of the parameter, to share same `KafkaProducer`.